lib/collection/src/wal_delta.rs

```diff
@@ -166,26 +158,32 @@ fn resolve_wal_delta(
 
     // If the recovery point has clocks our current node does not know about
     // we're missing essential records and cannot resolve a WAL delta
-    if recovery_point.has_clocks_not_in(local_recovery_point) {
+    if recovery_point.has_clocks_not_in(&local_recovery_point) {
         return Err(WalDeltaError::UnknownClocks);
     }
 
     // If our current node has any lower clock than the recovery point specifies,
     // we're missing essential records and cannot resolve a WAL delta
-    if recovery_point.has_any_higher(local_recovery_point) {
+    if recovery_point.has_any_higher(&local_recovery_point) {
         return Err(WalDeltaError::HigherThanCurrent);
     }
 
+    // From this point, increase all clocks by one
+    // We must do that so we can specify clock tick 0 as needing everything from that clock
+    recovery_point.increase_by(1);
+    local_recovery_point.increase_by(1);
+    local_cutoff_point.increase_by(1);
```

Comment:
We now do +1 in a later stage of our WAL delta resolution logic. We need to do this so we can represent 0 as 'we need all clocks for this peer_id+clock_id, including tick 0'.

Without the +1 here, using 0 would mean we'd only get ticks 1 and up, missing tick 0.

Doing the +1 here is much more localized, and only internal to this function. That is a lot better than making the whole recovery point off-by-one by default like we had before.